pyqtgraph/graphicsItems/LegendItem.py

```diff
@@ -218,6 +221,10 @@ def addItem(self, item, name):
             sample = item
         else:
             sample = self.sampleType(item)
+
+        if hasattr(sample, 'sigClicked'):
+            sample.sigClicked.connect(self.sampleClickEvent)
```

Comment:
There is no need for the sampleClickEvent method, you can connect one signal directly to another

```python
sample.sigClicked.connect(self.sigSampleClicked)
```